src/project/project_controller.ts

```diff
@@ -0,0 +1,213 @@
+import { Router, Request, Response, NextFunction,} from 'express';
+import Controller from '../interface/controller_interface';
+import projectModel from './project_model';
+import { Project } from './project_interface';
+import { createProjectSchema, updateProjectStatusSchema, addTeamLeadProjectSchema, addOrRemoveTeamMemberProjectSchema ,validateBody } from '../middleware/validateBody';
+import { authorizeAdmin, authorizeProjectManager, authorizeTeamLead, authorizeTeamMember, authorizeUser } from '../middleware/authorization';
+import { RequestWithUser } from 'interface/requestWithUser';
+
+class ProjectController implements Controller{
+    
+    path: string = '/project';
+    router: Router = Router();
+    private project = projectModel;
+
+    constructor() {
+        this.initializeRoutes()
```

Comment:
Every route uses `authorizeUser` so it's better to call it here.
```suggestion
        this.router.use(authorizeUser);
        this.initializeRoutes()
```
This way we do not need to use it every route.